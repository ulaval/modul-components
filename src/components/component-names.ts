export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
