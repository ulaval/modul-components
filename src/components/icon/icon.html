<svg class="m-icon" @click="onClick" @keydown="onKeydown" :aria-hidden="!hasSvgTitle" :width="size" :height="size">
    <title v-if="hasSvgTitle">{{svgTitle}}</title>
    <use :xlink:href="spriteId" aria-hidden="true"></use>
</svg>
