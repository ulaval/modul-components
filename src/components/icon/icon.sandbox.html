<div>
    <h3>Icône</h3>
    <h4>Propriété name vide</h4>
    <p>
        <m-icon></m-icon>
    </p>
    <h4>Propriété name invalide</h4>
    <p>
        <m-icon name="1234"></m-icon>
    </p>
    <h4>Propriété name courte</h4>
    <p>
        <m-icon name="edit"></m-icon>
    </p>
    <h4>Propriété name original</h4>
    <p>
        <m-icon name="m-svg__edit"></m-icon>
    </p>
    <h3>Pastille</h3>
    <h4>États de la pastille</h4>
    <p>
        <h5>Complété (completed)</h5>
        <m-icon-file v-m-badge="{ state: 'completed' }" extension="pdf"></m-icon-file>
        <h5>En erreur (error)</h5>
        <m-icon-file v-m-badge="{ state: 'error' }" extension="pdf"></m-icon-file>
        <h5>Avertissement (warning)</h5>
        <m-icon-file v-m-badge="{ state: 'warning' }" extension="pdf"></m-icon-file>
    </p>

    <h4>Tailles</h4>
    <p>
        <m-icon name="m-svg__calendar" v-m-badge="{ state: 'completed'}" size="12px"></m-icon>
        <m-icon name="m-svg__calendar" v-m-badge="{ state: 'completed'}" size="24px"></m-icon>
        <m-icon name="m-svg__calendar" v-m-badge="{ state: 'completed'}" size="48px"></m-icon>
        <m-icon name="m-svg__calendar" v-m-badge="{ state: 'completed'}" size="72px"></m-icon>
        <m-icon name="m-svg__calendar" v-m-badge="{ state: 'completed'}" size="144px"></m-icon>
    </p>

    <h4>Ajustement de la position de la pastille</h4>
    L'icône d'horloge n'a pas de position définie par le concepteur, par défaut, la pastille est placé en bas à droite du carré de 24px par 24px. Il est possible d'ajuster sa position.
    <p>
        <h5>Par défaut</h5>
        <m-icon name="m-svg__clock" v-m-badge="{ state: 'completed'}" size="24px"></m-icon>
        <br/>
        <m-icon name="m-svg__clock" v-m-badge="{ state: 'completed'}" size="72px"></m-icon>
        <h5>Après</h5>
        size: 24px
        offsetX: -2px
        offsetY: -1px
        <br/>
        <m-icon name="m-svg__clock" v-m-badge="{ state: 'completed', offsetX: '-2px', offsetY: '-1px'}" size="24px"></m-icon>
        <br/>
        <br/>
        size: 72px
        offsetX: -6px
        offsetY: -3px
        <br/>
        <m-icon name="m-svg__clock" v-m-badge="{ state: 'completed', offsetX: '-6px', offsetY: '-3px'}" size="72px"></m-icon>
    </p>
</div>
