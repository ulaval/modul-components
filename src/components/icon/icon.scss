.m-icon {
    display: inline-flex;
    align-items: center;
    justify-content: center;
    fill: currentColor;
    outline: none;
}
