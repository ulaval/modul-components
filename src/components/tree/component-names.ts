export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
