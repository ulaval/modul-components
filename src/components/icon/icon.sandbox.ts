import Vue, { PluginObject } from 'vue';
import { Component } from 'vue-property-decorator';

import { ICON_NAME } from '../component-names';
import WithRender from './icon.sandbox.html';

@WithRender
@Component
export class MIconSandbox extends Vue {
}

const IconSandboxPlugin: PluginObject<any> = {
    install(v, options): void {
        v.component(`${ICON_NAME}-sandbox`, MIconSandbox);
    }
};

export default IconSandboxPlugin;
