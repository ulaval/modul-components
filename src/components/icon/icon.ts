import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './icon.html?style=./icon.scss';
import { ICON_NAME } from '../component-names';

@WithRender
@Component
export class MIcon extends Vue {
    @Prop({ default: 'default' })
    public name: string;
    @Prop()
    public svgTitle: string;
    @Prop({ default: '1em' })
    public size: string;

    protected beforeMount(): void {
        if (!document.getElementById(this.name)) {
            console.warn('"' + this.name + '" is not a valid svg id. Make sure that the sprite has been loaded via the $svg instance service.');
        }
    }

    private get hasSvgTitle(): boolean {
        return !!this.svgTitle;
    }

    private get spriteId(): string {
        return '#' + this.name;
    }

    private onClick(event): void {
        this.$emit('click', event);
    }

    private onKeydown(event): void {
        this.$emit('keydown', event);
    }
}

const IconPlugin: PluginObject<any> = {
    install(v, options) {
        console.debug(ICON_NAME, 'plugin.install');
        v.component(ICON_NAME, MIcon);
    }
};

export default IconPlugin;
