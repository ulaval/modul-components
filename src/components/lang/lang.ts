import Vue from 'vue';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './lang.html';
import { I18N_NAME } from '../component-names';
import { ModulVue } from '../../utils/vue';

@WithRender
@Component
export class MLang extends ModulVue {
    public text: string = '';

    @Prop()
    public msg: string;

    public mounted() {
        this.text = this.$i18n.translate(this.msg);
    }
}

const LangPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(I18N_NAME, MLang);
    }
};

export default LangPlugin;
