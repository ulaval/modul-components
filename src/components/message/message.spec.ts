import { createLocalVue, mount } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';

import { addMessages } from '../../../tests/helpers/lang';
import { renderComponent } from '../../../tests/helpers/render';
import { ICON_NAME } from '../component-names';
import { MIcon } from '../icon/icon';
import MessagePlugin, { MMessage, MMessageSkin, MMessageState } from './message';

describe('MMessage', () => {
    let localVue: VueConstructor<Vue>;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(MessagePlugin);
        addMessages(localVue, ['components/message/message.lang.en.json']);
    });

    it('should render correctly', () => {
        const msg = mount(MMessage, {
            localVue: localVue
        });

        return expect(renderComponent(msg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly content', () => {
        const msg = mount(MMessage, {
            localVue: localVue,
            slots: {
                default: 'message'
            }
        });

        return expect(renderComponent(msg.vm)).resolves.toMatchSnapshot();
    });

    it('should render nothing if not visible', () => {
        const msg = mount(MMessage, {
            localVue: localVue,
            propsData: {
                visible: false
            }
        });

        return expect(renderComponent(msg.vm)).resolves.toEqual('');
    });

    it('should render correctly light skin', () => {
        const msg = mount(MMessage, {
            localVue: localVue,
            propsData: {
                skin: MMessageSkin.Light
            }
        });

        return expect(renderComponent(msg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly all possible states', async () => {
        const btn = mount(MMessage, {
            localVue: localVue
        });

        for (const state in MMessageState) {
            btn.setProps({ state: MMessageState[state] });
            expect(await renderComponent(btn.vm)).toMatchSnapshot(state);
        }
    });

    it('should render correctly when there is no icon', () => {
        const msg = mount(MMessage, {
            localVue: localVue,
            propsData: {
                icon: false
            }
        });

        return expect(renderComponent(msg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly close button when prop is set', () => {
        // For vue-test-utils bug, else m-icon is not rendered in m-button-icon
        Vue.component(ICON_NAME, MIcon);

        const msg = mount(MMessage, {
            localVue: localVue,
            propsData: {
                closeButton: true
            }
        });

        return expect(renderComponent(msg.vm)).resolves.toMatchSnapshot();
    });

    it('should render nothing after close button is clicked', async () => {
        const msg = mount(MMessage, {
            localVue: localVue,
            propsData: {
                closeButton: true
            }
        });

        msg.find('button').trigger('click');
        return expect(renderComponent(msg.vm)).resolves.toEqual('');
    });

    it('should emit close event when close button is clicked', () => {
        const msg = mount(MMessage, {
            localVue: localVue,
            propsData: {
                closeButton: true
            }
        });

        msg.find('button').trigger('click');

        expect(msg.emitted('close')).toBeTruthy();
    });
});
