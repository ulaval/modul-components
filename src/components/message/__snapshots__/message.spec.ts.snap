// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MMessage should render correctly 1`] = `
<div class="m-message m--is-success m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#check" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly all possible states: Error 1`] = `
<div class="m-message m--has-error m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#error" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly all possible states: Information 1`] = `
<div class="m-message m--is-information m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#information" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly all possible states: Success 1`] = `
<div class="m-message m--is-success m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#check" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly all possible states: Warning 1`] = `
<div class="m-message m--is-warning m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#warning" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly close button when prop is set 1`] = `
<div class="m-message m--is-success m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#check" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body m--has-close-button">
            <button type="button" class="m-icon-button m-message__body__close-button m--is-skin-light" style="width:44px;height:44px;font-size:10px;">
                <svg aria-hidden="true" width="1em" height="1em" class="m-icon m-icon-button__icon">
                    <use xlink:href="#close" aria-hidden="true"></use>
                </svg> <span class="m-icon-button__hidden"><span class="m-i18n">Close</span></span>
            </button>
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly content 1`] = `
<div class="m-message m--is-success m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#check" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content">message</div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly light skin 1`] = `
<div class="m-message m--is-success m--is-skin-light">
    <div class="m-message__wrap">
        <div class="m-message__icon">
            <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                <use xlink:href="#check" aria-hidden="true"></use>
            </svg>
        </div>
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;

exports[`MMessage should render correctly when there is no icon 1`] = `
<div class="m-message m--is-success m--is-skin-regular">
    <div class="m-message__wrap">
        <div class="m-message__body">
            <div class="m-message__body__content"></div>
        </div>
    </div>
</div>
`;
